src/App.tsx

```diff
@@ -14,11 +15,77 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  isVisible: boolean,
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
```

Comment:
It's better to use a copy of `goodsFromServer`
```suggestion
    goods: [...goodsFromServer],
```